src/imitation/util/networks.py

```diff
@@ -165,23 +172,35 @@ def update_stats(self, batch: th.Tensor) -> None:
             batch: A batch of data to use to update the running mean and variance.
         """
         b_size = batch.shape[0]
+        reduce_dims = [0, 2, 3] if len(batch.shape) == 4 else 0
 
         if self.count == 0:
-            self.running_mean = th.mean(batch, dim=0)
+            self.running_mean = th.mean(batch, dim=reduce_dims)
             if b_size > 1:
-                self.running_var = th.var(batch, dim=0)
+                self.running_var = th.var(batch, dim=reduce_dims)
         else:
             # Shuffle the batch since we don't don't want to bias the mean
-            # towards data that appears latter in the batch
+            # towards data that appears later in the batch
             perm = th.randperm(b_size)
 
             alpha = 1 - self.decay
 
             for i in range(b_size):
-                diff = batch[perm[i], ...] - self.running_mean
+                new_sample = (
+                    th.mean(batch[perm[i], ...], dim=[1, 2])
+                    if len(batch.shape) == 4
+                    else batch[perm[i], ...]
+                )
+                diff = new_sample - self.running_mean
                 incr = alpha * diff
                 self.running_mean += incr
                 self.running_var = self.decay * (self.running_var + diff * incr)
+                if len(batch.shape) == 4:
```

Comment:
Why is this needed?

Reply:
Basically to account for within-image variance. Imagine each image is identical, with pixel values uniformly distributed between 0 and 255. If we just add `new_sample - self.running_mean` terms to `self.running_var`, then `self.running_var` would be 0.

Reply:
I can type up the proof that this works if we end up keeping this style of normalization.